src/NServiceBus.AcceptanceTests/Core/UnitOfWork/When_using_a_func_for_UOW_scope.cs

```diff
@@ -0,0 +1,98 @@
+﻿namespace NServiceBus.AcceptanceTests.Core.UnitOfWork
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NUnit.Framework;
+
+    public class When_using_a_func_for_UOW_scope : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_only_create_one_instance_of_UnitOrWorkInjected_per_message()
+        {
+            Func<ObjectBuilder.IResolver, ToBeInjected> func = v =>
+            {
+                var uow = v.Build<UnitOrWorkInjected>();
+                return new ToBeInjected(uow);
+            };
+
+            // ReSharper disable once UnusedVariable
+            var context = await Scenario.Define<Context>()
+                .WithEndpoint<Endpoint>(b =>
+                {
+                    b.CustomConfig(c => c.RegisterComponents(r =>
+                    {
+                        r.ConfigureComponent<Context>(DependencyLifecycle.SingleInstance);
+
+                        r.ConfigureComponent<UnitOrWorkInjected>(DependencyLifecycle.InstancePerUnitOfWork);
+                        r.ConfigureComponent(func, DependencyLifecycle.InstancePerUnitOfWork);
+                    }));
+
+                    b.When(async (bus, c) =>
+                    {
+                        await bus.SendLocal(new Endpoint.MyMessage());
+                        await bus.SendLocal(new Endpoint.MyMessage());
+                        await bus.SendLocal(new Endpoint.MyMessage());
+                    });
+                })
+                .Done(c => c.CalledCount >= 3)
+                .Run(TimeSpan.FromSeconds(15));
+
+            Assert.AreEqual(3, UnitOrWorkInjected.InstanceCount);
```

Comment:
I don't agree, the name of the test matches with sending 3 separate messages which should result in 3 UOW scopes and therefore 3 separate instances.

Multiple handlers per message already resulted in the same instance anyway